io/io/src/TFileMerger.cxx

```diff
@@ -892,8 +955,11 @@ Bool_t TFileMerger::PartialMerge(Int_t in_type)
    } else {
       // Close or write is required so the file is complete.
       if (in_type & kIncremental) {
-         fOutputFile->Write("",TObject::kOverwrite);
+         if (!(in_type & kDelayWrite))
+            fOutputFile->Write("",TObject::kOverwrite);
```

Comment:
Added:
```
         // In the case of 'kDelayWrite' the caller want to avoid having to 
         // write the output objects once for every input file and instead
         // write it only once at the end of the process.
```